packages/keymaps/src/common/user-storage-watcher.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { Disposable, DisposableCollection, Emitter, Event } from '@theia/core/lib/common';
+import { UserStorageChangeEvent, StorageServer } from './user-storage-protocol';
+
+
+@injectable()
+export class UserStorageWatcher implements Disposable {
```

Comment:
I thought that watchers are redundant since we can call `server.setClient` anywhere.

Reply:
I tried that before and weirdly enough only events coming from outside of theia worked. I thought it was the server not being a singleton but even with this modifying from within theia always failed. Maybe something got disposed too quickly?